intake/__init__.py

```diff
@@ -27,8 +27,10 @@
 # Populate list of autodetected drivers (plugins).
 # The built-in ones are included here because intake itself declares them via
 # its own entry_points.
-for name, driver in autodiscover().items():
-    register_driver(name, driver)
+with warnings.catch_warnings():
+    warnings.simplefilter("ignore")
+    for name, driver in autodiscover().items():
+        register_driver(name, driver)
```

Comment:
What drivers were warning here? Emitting this warning is in our control, the idea was to remove this version of autodiscover (as opposed to entryoints) eventually.

Reply:
The warning is coming from

https://github.com/intake/intake/blob/76cb2ff2772bf56720fcefcbde5e0e2eb75401a6/intake/source/discovery.py#L55-L60

and is independent of any particular driver.

Reply:
Hm, OK - probably should only warn if we actually find anything

Reply:
I went with the minimal fix to prevent warning on import in down stream packages.

Reply:
I know, but this basically makes the warning always a no-op; would you mind removing the ignore, and only issue the warnings of the package scan returns anything?

Reply:
This only skips the warning on import, if the user manually calls `autodiscover()` it will still warn them unless they explicitly pass `do_package_scan=False`.  It seems correct to me to warn the user if something _may_ have been found given their call even if it was not.

Reply:
The purpose of the warning is different: that the user has a package installed which uses the deprecated method to be found by intake. That *does* happen at install time, only; I never expected any user to call the function directly.

Reply:
I see.  So the remediation is to switch the package to using entry points instead?

Reply:
Yes, this is supposed to get users to switch to a new version of the package containing the driver, and to get the dev of the package to update it to use entrypoints (not in that order!)

Reply:
Would you mind reinstating the warning (i.e., dropping the ignore), but making it trigger only if the package scan turns up something?